### ISSUE TITLE
Stack trace using windows/meterpreter/reverse_hop_http

### ISSUE THREAD

**Issue by Kanareika** [rapid7/metasploit-framework#5396]
I'm using generated payload windows/metepreter/reverse_hop_http, and meterpreter shell is not working. Log here. Metasploit updated by msfupdate and apt-get upgrade.

```
msf exploit(handler) > exploit -j
[*] Exploit running as background job.

[*] Starting the payload handler...
msf exploit(handler) > [*] Preparing stage for next session 47EgK_4TC4Wl4EgFUufL8e
[*] Uploaded stage to hop http:/*******/hop.php
[*] Meterpreter session 1 opened (Hop client -> **********:80) at 2015-05-21 15:59:22 +0500
[*] Preparing stage for next session FSJ9F_CU5XddDfA16pFwbs
[*] Uploaded stage to hop *********hop.php?/
[-] Error while running command load: undefined method `type?' for #<#<Class:0xc932ea0>:0x10f9fdd4>
```

Call stack:
```
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:74:in `commands'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:387:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/usr/share/metasploit-framework/lib/msf/base/sessions/meterpreter.rb:287:in `load_stdapi'
/usr/share/metasploit-framework/lib/msf/base/sessions/meterpreter_options.rb:57:in `block in on_session'
/usr/share/metasploit-framework/lib/msf/core/session_manager.rb:171:in `call'
/usr/share/metasploit-framework/lib/msf/core/session_manager.rb:171:in `block (2 levels) in initialize_scheduler_threads'
/usr/share/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `call'
/usr/share/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `block in spawn'
[-] Error while running command load: undefined method `type?' for #<#<Class:0xc932ea0>:0x10f9fdd4>
```

Call stack:
```
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:74:in `commands'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:387:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/usr/share/metasploit-framework/lib/msf/base/sessions/meterpreter.rb:298:in `load_priv'
/usr/share/metasploit-framework/lib/msf/base/sessions/meterpreter_options.rb:64:in `block in on_session'
/usr/share/metasploit-framework/lib/msf/core/session_manager.rb:171:in `call'
/usr/share/metasploit-framework/lib/msf/core/session_manager.rb:171:in `block (2 levels) in initialize_scheduler_threads'
/usr/share/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `call'
/usr/share/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `block in spawn'
```

Okay, maybe it's work.
```
msf exploit(handler) > sessions -i 1
[*] Starting interaction with 1...

meterpreter > getuid
[-] Error while running command getuid: undefined method `type?' for #<#<Class:0xc932ea0>:0x10f9fdd4>
```

Call stack:
```
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:74:in `commands'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:387:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console.rb:68:in `block in interact'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:193:in `call'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:193:in `run'
/usr/share/metasploit-framework/lib/rex/post/meterpreter/ui/console.rb:66:in `interact'
/usr/share/metasploit-framework/lib/msf/base/sessions/meterpreter.rb:431:in `_interact'
/usr/share/metasploit-framework/lib/rex/ui/interactive.rb:49:in `interact'
/usr/share/metasploit-framework/lib/msf/ui/console/command_dispatcher/core.rb:1854:in `cmd_sessions'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:427:in `run_command'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:389:in `block in run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `each'
/usr/share/metasploit-framework/lib/rex/ui/text/dispatcher_shell.rb:383:in `run_single'
/usr/share/metasploit-framework/lib/rex/ui/text/shell.rb:203:in `run'
/usr/share/metasploit-framework/lib/metasploit/framework/command/console.rb:38:in `start'
/usr/share/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
/opt/metasploit/apps/pro/msf3/msfconsole:48:in `<main>'
```

No, it's not work.

**Comment by coldalfred** [comment]
UUID hash the lastes meterpreters bin change the magic the initial request was /XxXx and now its 256 bytes i think i dont know much about it

If you need this payload to work quickly you can revert the metasploit to an older version and test it.

Its work for me the version of the lastest kali ISO that come default.

**Comment by hmoore-r7** [comment]
Looks like ```reverse_http_hop``` is still using the old URL format. Thanks!

**Comment by bcook-r7** [comment]
@hmoore-r7 do you need any assistance here?

**Comment by hmoore-r7** [comment]
Just the reminder, thanks!

**Comment by hmoore-r7** [comment]
This one is a bit of a mess and will take substantial effort to refactor

**Comment by bcook-r7** [comment]
For reference, the original PR is here: https://github.com/rapid7/metasploit-framework/pull/2809/files

**Comment by bcook-r7** [comment]
This is fixed in #5990, thanks.
